### PR TITLE
[API-39422] Add `DependentClaimantVerification` concern to v1

### PR DESCRIPTION
**Pull request by tycol7**
## Summary

Create a `DependentClaimantVerification` concern that takes a participant id and dependent name then calls BGS to verify that the dependent name belongs to the participant.

Add the verification to the v1 Power of Attorney `/validate` endpoint behind a feature flag.

## Related issue(s)

[API-39422](https://jira.devops.va.gov/browse/API-39422)

## Testing done

- [x] *New code is covered by unit tests*

## Screenshots

N/A

## What areas of the site does it impact?

- v1 Power of Attorney `/validate` endpoint
- new concern

## Acceptance criteria

- [x]  I added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

We don't know the exact shape of the response from BGS if we get multiple dependents back. Interested to see how we can handle this unknown or if we can get an example response.

### CODE REVIEW

File: modules/claims_api/app/controllers/concerns/claims_api/dependent_claimant_verification.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    def valid_participant_dependent_combo?(participant_id, dependent_first_name, dependent_last_name)
+      return false if participant_id.blank?
+
+      person_web_service = PersonWebService.new(external_uid: 'dependent_claimant_verification_uid',
+                                                external_key: 'dependent_claimant_verification_key')
+      dependents = person_web_service.find_dependents_by_ptcpnt_id(participant_id)
+
+      return false if dependents.nil? || dependents[:number_of_records].to_i.zero?
+
+      Array.wrap(dependents).any? do |dependent|
```
Comment (tycol7) [inline modules/claims_api/app/controllers/concerns/claims_api/dependent_claimant_verification.rb:36]: see: https://apidock.com/rails/Array/wrap/class

we know we [get an object response](https://github.com/department-of-veterans-affairs/vets-api/blob/ec36fe81aed483adb69a3b1f8a3cea666bd18542/modules/claims_api/spec/lib/claims_api/person_web_service_spec.rb) if there is one dependent. we don't know if we get an object or array if there are multiple dependents. wrap should handle both cases.

n.b. we assume each dependent will have the `dependent` key at the root level in the "many" case as it does in the "singular" case. might not be true.

File: modules/claims_api/app/controllers/concerns/claims_api/dependent_claimant_verification.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    end
+
+    def valid_participant_dependent_combo?(participant_id, dependent_first_name, dependent_last_name)
+      return false if participant_id.blank?
+
+      person_web_service = PersonWebService.new(external_uid: 'dependent_claimant_verification_uid',
+                                                external_key: 'dependent_claimant_verification_key')
+      dependents = person_web_service.find_dependents_by_ptcpnt_id(participant_id)
+
+      return false if dependents.nil? || dependents[:number_of_records].to_i.zero?
```
Comment (tycol7) [inline modules/claims_api/app/controllers/concerns/claims_api/dependent_claimant_verification.rb:34]: we get `{:number_of_records=>"0"}` if there are no dependents. not sure if we can get back other bad responses - open to suggestions on how to make this guard more robust.

File: modules/claims_api/app/controllers/concerns/claims_api/dependent_claimant_verification.rb (hunk 1 above)
Comment (tycol7) [reply modules/claims_api/app/controllers/concerns/claims_api/dependent_claimant_verification.rb:39]: we found a participant with multiple dependents. turns out there is one `dependent` key that contains an array. thanks Rocky! https://github.com/department-of-veterans-affairs/vets-api/pull/18416/commits/19811d35469c941e60f7d8341d5e670d8095295d

**Review by rockwellwindsor-va — APPROVED** [review]
Ran this with Janet and it behaved as expected with those test users and their data.  So this LGTM!

**Pull request merged**